mne/simulation/evoked.py

```diff
@@ -69,8 +69,14 @@ def simulate_evoked(fwd, stc, info, cov, snr=3., tmin=None, tmax=None,
     evoked = apply_forward(fwd, stc, info)
     if snr < np.inf:
         noise = simulate_noise_evoked(evoked, cov, iir_filter, random_state)
-        evoked_noise = add_noise_evoked(evoked, noise, snr,
-                                        tmin=tmin, tmax=tmax)
+        if evoked.data.shape != noise.data.shape:
```

Comment:
This check should probably be in `add_noise_evoked` somewhere, and also it should probably check that the set of channels is the same, not just that the data sizes are the same